### PR TITLE
UPSTREAM: 63875: make TestGetServerGroupsWithTimeout more reliable

### PR DESCRIPTION
**Pull request by deads2k**
Hopefully fixes the unit test flake we're seeing.

/assign @soltysh

### CODE REVIEW

**Comment by deads2k** [comment]
Addresses https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/19692/test_pull_request_origin_unit/13062/

**Comment by mfojtik** [comment]
wait for tests to be green, LGTM otherwise

**Comment by deads2k** [comment]
test unit is green, this only touches unit tests.  manually merging because this flake is crazy common.

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go [hunk 1]
```diff
@@ -143,18 +143,18 @@ func TestGetServerGroupsWithTimeout(t *testing.T) {
 	}))
 	defer server.Close()
 	defer close(done)
-	tmp := defaultTimeout
-	defaultTimeout = 2 * time.Second
```
Comment (soltysh) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go]: That here was intentionally to ensure the default is applied out of the box. Without it, there's nothing checking that it is set. I'm ok with extending the error part, but this bit should be brought back.

Comment (soltysh) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go]: Let's sync tomorrow, I'd like to re-shape this PR before merging upstream.

Comment (deads2k) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go]: > That here was intentionally to ensure the default is applied out of the box. Without it, there's nothing checking that it is set. I'm ok with extending the error part, but this bit should be brought back.

We need to check that the feature works, not that a bit of wiring does.  Checking that wiring works is simply duplicating code from the original.  And global state mutation in a test is a bad pattern.

Comment (soltysh) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go]: How else you want to ensure the wiring is happening, esp. if it applies only for the dynamic client, but not for regular one? I do agree that global state mutation in tests is a bad pattern, but it's quite frequent all over our code base.

Comment (deads2k) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go]: > How else you want to ensure the wiring is happening, esp. if it applies only for the dynamic client, but not for regular one?

To be honest, I don't much care.  If we have a problem we'll chase it and have an easy spot to wire up the fix.  If the functionality works and we're short wiring a default, it's fairly easy to figure out how to connect the bits.  Testing wiring is often a duplication of the code its testing.

Comment (soltysh) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/discovery/discovery_client_test.go]: Discussed this in depth with David on IRC, I'm ok with this approach.

**Pull request merged**